pkg/controller/controller_binding_test.go

```diff
@@ -904,6 +977,124 @@ func TestReconcileBindingWithBrokerHTTPError(t *testing.T) {
 	}
 }
 
+func TestReconcileBindingWithFailureCondition(t *testing.T) {
```

Comment:
For each of these tests that you're adding, can you make sure they meet all the criteria in #1077